src/xenia/kernel/xam/xam_input.cc

```diff
@@ -118,6 +120,10 @@ dword_result_t XamInputSetState(dword_t user_index, dword_t unk,
     actual_user_index = 0;
   }
 
+  if (!cvars::enable_vibration) {
+    vibration->left_motor_speed = 0;
```

Comment:
This must not be modifying the original structure in the guest memory, it should be creating a copy in this case.

Reply:
+1, just "allocate" a new one on the stack (in the function scope)
You could also have the memcpy and pointer assignment inside the if block

Also, i suggest moving this code to `InputSystem::SetState`